service/api.go

```diff
@@ -106,14 +107,27 @@ func (s *gnomobileService) SelectAccount(ctx context.Context, req *connect.Reque
 	}
 
 	s.lock.Lock()
-	s.activeAccount = key
+	account, ok := s.userAccounts[req.Msg.NameOrBech32]
+	if !ok {
+		account = &userAccount{}
+		s.userAccounts[req.Msg.NameOrBech32] = account
+	}
+	account.keyInfo = key
+	s.activeAccount = account
 	s.lock.Unlock()
 
 	s.getSigner().Account = req.Msg.NameOrBech32
+	s.getSigner().Password = account.password
 	return connect.NewResponse(&rpc.SelectAccountResponse{Key: info}), nil
 }
 
 func (s *gnomobileService) SetPassword(ctx context.Context, req *connect.Request[rpc.SetPasswordRequest]) (*connect.Response[rpc.SetPasswordResponse], error) {
+	s.lock.Lock()
+	if s.activeAccount != nil {
```

Comment:
Should we accept to set password if no active account is set?
What is the flow in the UI, can we change the password of any account without the need of selecting them as active account?
If yes, so we should give as argument for which account we are setting the password in the account map.

Reply:
> Should we accept to set password if no active account is set?

You're right. Maybe it should return ErrCode_ErrNoActiveAccount?

> Can we change the password of any account without the need of selecting them as active account?

I think that would be confusing. Other methods apply to the active account, such as Call. But we can discuss.

Reply:
> > Should we accept to set password if no active account is set?
> 
> You're right. Maybe it should return ErrCode_ErrNoActiveAccount?

Yes.
> 
> > Can we change the password of any account without the need of selecting them as active account?
> 
> I think that would be confusing. Other methods apply to the active account, such as Call. But we can discuss.

Yes, I agree to do actions only on the active account.

Reply:
Please see the updated code and check my use of "defer".
https://github.com/gnolang/gnomobile/blob/eb084d42a683769a9bbe9ef3b6e10a916e4c7848/service/api.go#L125-L129

Reply:
Yes you can use the `defer` here I think because you don't call any long-time operation after.